src/storage/database/knex.ts

```diff
@@ -467,12 +467,15 @@ export class StorageKnexDB implements Database {
       tnx.on('query-error', (error: DatabaseError) => {
         throw DBError.fromDBError(error)
       })
-      await this.connection.setScope(tnx)
-    } else if (differentScopes) {
-      await this.connection.setScope(tnx)
     }
 
     try {
+      if (needsNewTransaction) {
+        await this.connection.setScope(tnx)
+      } else if (differentScopes) {
+        await this.connection.setScope(tnx)
+      }
```

Comment:
Maybe it's more concise and/or readable if it's set like this?

```suggestion
      if (needsNewTransaction || differentScopes) {
        await this.connection.setScope(tnx)
      }
```

Reply:
Way more readable, good catch